dpctl-capi/tests/test_sycl_device_interface.cpp

```diff
@@ -255,6 +255,17 @@ TEST_P(TestDPCTLSyclDeviceInterface, Chk_IsHost)
     EXPECT_NO_FATAL_FAILURE(DPCTLDevice_Delete(DRef));
 }
 
+TEST_P(TestDPCTLSyclDeviceInterface, Chk_CreateSubDevicesEqually)
+{
+    DPCTLSyclDeviceRef DRef = nullptr;
```

Comment:
How about something like this? The reason I have started wrapping `EXPECT_NO_FATAL_FAILURE` is to make sure we are always catching exceptions and not crashing.

You will also need to include `#include "dpctl_sycl_device_manager.h"` in this file.
```
    DPCTLSyclDeviceRef DRef = nullptr;
    DPCTLDeviceVectorRef DVRef = nullptr;
    uint32_t maxCUs = 0;

    EXPECT_NO_FATAL_FAILURE(DRef = DPCTLDevice_CreateFromSelector(DSRef));
    if (!DRef)
        GTEST_SKIP_("Device not found");

    EXPECT_NO_FATAL_FAILURE(maxCUs = DPCTLDevice_GetMaxComputeUnits(DRef));
    if (maxCUs) {
        int count = maxCUs / 2;
        EXPECT_NO_FATAL_FAILURE(
            DVRef = DPCTLDevice_CreateSubDevicesEqually(DRef, count));
        if (DVRef) {
            EXPECT_TRUE(DPCTLDeviceVector_Size(DVRef) > 0);
            EXPECT_NO_FATAL_FAILURE(DPCTLDeviceVector_Delete(DVRef));
        }
    }

    EXPECT_NO_FATAL_FAILURE(DPCTLDevice_Delete(DRef));
```